### PR TITLE
PYIC-6917 Change made to store featureSet with CriResponseItem to be used in async lambda for f2f.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Implement featuresets for async CRI callback

### What changed
Change made to store featureSet with CriResponseItem to be used in async lambda for f2f.

### Issue tracking

- [PYIC-6917](https://govukverify.atlassian.net/browse/PYIC-6917)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-6917]: https://govukverify.atlassian.net/browse/PYIC-6917?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/cri-storing-service/src/main/java/uk/gov/di/ipv/core/library/cristoringservice/CriStoringService.java [hunk 1]
```diff
@@ -79,6 +79,7 @@ public void recordCriResponse(
                 journeyRequest.getIpvSessionId(),
                 journeyRequest.getIpAddress(),
                 journeyRequest.getDeviceInformation(),
+                null,
```
Comment (Joe-Edwards-GDS) [inline libs/cri-storing-service/src/main/java/uk/gov/di/ipv/core/library/cristoringservice/CriStoringService.java]: Can we use `journeyRequest.getFeatureSet()`? (You might need to use `RequestHelper.getFeatureSet` to parse it)

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/CriResponseItem.java [hunk 2]
```diff
@@ -25,6 +26,7 @@ public class CriResponseItem implements DynamodbItem {
     private Instant dateCreated;
     private String status;
     private long ttl;
+    private List<String> featureSet;
```
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/CriResponseItem.java:29]: (I assume this works fine for existing records)

File: libs/cri-storing-service/src/main/java/uk/gov/di/ipv/core/library/cristoringservice/CriStoringService.java (hunk 1 above)
Comment (rpayal) [reply libs/cri-storing-service/src/main/java/uk/gov/di/ipv/core/library/cristoringservice/CriStoringService.java]: Done

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/CriResponseItem.java (hunk 2 above)
Comment (rpayal) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/CriResponseItem.java:29]: Tested with already existing f2f records in dev.

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**